For RFC5424 APP-NAME is limited to 48 chars - so truncate it

scorptec68 opened the pull request:
APP-NAME        = NILVALUE / 1*48PRINTUSASCII
Keep the last 48 chars as that is more interesting generally.
Could use elipsis truncating but then lose information - not sure if
worth it.

sirsean commented on formatter.go:
```diff
@@ -37,11 +37,19 @@ func RFC3164Formatter(p Priority, hostname, tag, content string) string {
 }
 
+// if string's length is greater than max, then use the last part
+func TruncateStartStr(s string, max int) string {
```
Let's make this private. `truncateStartStr` please.

sirsean commented on constants.go:
```diff
@@ -12,6 +12,7 @@ type Priority int
 
 const severityMask = 0x07
 const facilityMask = 0xf8
+const appNameMax = 48 // limit to 48 chars as per RFC5424
```
`appNameMaxChars` or `appNameMaxLength` is more descriptive, but I wonder if this would be better as a local value inside the `RFC5424Formatter` function (because it's RFC5424-specific).

scorptec68 commented:
I am just new to go-lang, so didn't know about not exporting methods using lowercase until now.
Hopefully, I have addressed your comments in the last commit. Please let me know otherwise.

sirsean commented:
No worries about being new. 😄 Thanks for the fix.

I might prefer that the `appNameMaxLength` constant indicate that it only applies to RFC5424, but that can be changed later if there's ever a different one.
